location.go

```diff
@@ -91,16 +91,20 @@ func init() {
 }
 
 // retrieve current location based on connection
-func getCurrentLocation(params map[string]*string) *time.Location {
+func (sc *snowflakeConn) getCurrentLocation() *time.Location {
+	if sc.location != nil {
+		return sc.location
+	}
 	loc := time.Now().Location()
 	var err error
 	paramsMutex.Lock()
-	if tz, ok := params["timezone"]; ok {
+	defer paramsMutex.Unlock()
+	if tz, ok := sc.cfg.Params["timezone"]; ok {
 		loc, err = time.LoadLocation(*tz)
 		if err != nil {
 			loc = time.Now().Location()
 		}
 	}
-	paramsMutex.Unlock()
```

Comment:
How to take care of ALTER SESSION command changes some parameter like TIMEZONE? will it be change in the cache? Could you please add test case for it?

Reply:
You are right. With your question in my mind, I started from scratch: https://github.com/snowflakedb/gosnowflake/pull/882